pkg/sidecar/sidecar_test.go

```diff
@@ -0,0 +1,139 @@
+package sidecar
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+const (
+	CONFIGUREDNETWORK   = "testtesttest_net"
+	CONFIGUREDBANDWIDTH = uint64(12345)
+	NETWORKDONE         = "configureddone"
+)
+
+func unique_runenv() (hostname string, runenv *runtime.RunEnv) {
+	unique := strconv.Itoa(rand.Int())
+	params := runtime.RunParams{
+		TestPlan:               "sidecarPlan" + unique,
+		TestCase:               "sidecarCase" + unique,
+		TestRun:                unique,
+		TestInstanceCount:      1,
+		TestInstanceRole:       "sidecarRole" + unique,
+		TestGroupID:            "sidecarGroup" + unique,
+		TestGroupInstanceCount: 1,
+	}
+	hostname = "sidecar.test." + unique
+	runenv = runtime.NewRunEnv(params)
+	return
+}
+
+func verifyDefaultNetwork(t *testing.T, n *MockNetwork) {
+	assert.Len(t, n.Configured, 1, "expected exactly one network to be configured")
+	def := n.Configured[0]
+	assert.Equal(t, "default", def.Network, "expected default network to be named `default`")
+}
+
+func verifyConfiguredNetwork(t *testing.T, n *MockNetwork) {
+	assert.Len(t, n.Configured, 2, "expected exactly two networks to be configured")
+	con := n.Configured[1]
+	assert.Equal(t, CONFIGUREDNETWORK, con.Network, "expected configured network name to be passed to network driver")
+	assert.Equal(t, CONFIGUREDBANDWIDTH, con.Default.Bandwidth, "expected requested bandwidth to be set")
+}
+
+// planRequestNetworkConfigure is a testground plan.
+// It requests a change to the network configuration via the sync service.
+func planRequestNetworkConfigure(ctx context.Context, runenv *runtime.RunEnv, hostname string, done chan int, doConfigure bool) (err error) {
+	client, err := sync.NewBoundClient(ctx, runenv)
+	if err != nil {
+		return
```

Comment:
```suggestion
		return err
```

Nitpick: I see that `err` is named, but just `return` doesn't really convey that you return the error.